server/proxy/pf_server.c

```diff
@@ -0,0 +1,432 @@
+ *
+ * The server may start sending graphics output and receiving keyboard/mouse
+ * input after this callback returns.
+ */
+BOOL pf_server_post_connect(freerdp_peer* client)
+{
+	proxyConfig* config;
+	pServerContext* ps;
+	pClientContext* pc;
+	HANDLE connectionClosedEvent;
+	proxyData* pdata;
+	char* host = NULL;
+	DWORD port = 3389; /* default port */
+	ps = (pServerContext*)client->context;
+	pdata = ps->pdata;
+	config = pdata->config;
+
+	if (config->UseLoadBalanceInfo)
+	{
+		if (!pf_server_parse_target_from_routing_token(client, &host, &port))
+		{
+			WLog_ERR(TAG, "pf_server_parse_target_from_routing_token failed!");
+			return FALSE;
+		}
+
+		WLog_DBG(TAG, "Parsed target from load-balance-info: %s:%i", host, port);
+	}
+	else
+	{
+		/* use hardcoded target info from configuration */
+		host = _strdup(config->TargetHost);
+		port = config->TargetPort > 0 ? config->TargetPort : port;
+		WLog_DBG(TAG, "Using hardcoded target host: %s:%i", host, port);
+	}
+
+	/* Start a proxy's client in it's own thread */
+	pc = (pClientContext*) p_client_context_create(client->settings, host, port);
+	pc->pdata = ps->pdata;
+	/* Inject proxy's client context to proxy's context */
+	connectionClosedEvent = CreateEvent(NULL, TRUE, FALSE, NULL);
+	pdata->pc = (pClientContext*) pc;
+	pdata->ps = ps;
+	pdata->connectionClosed = connectionClosedEvent;
+	pf_server_rdpgfx_init(ps);
+
+	if (!(ps->thread = CreateThread(NULL, 0, pf_client_start, (rdpContext*) pc, 0,
+	                                NULL)))
+	{
+		WLog_ERR(TAG, "CreateThread failed!");
+		return FALSE;
+	}
+
+	return TRUE;
+}
+
+BOOL pf_server_activate(freerdp_peer* client)
+{
+	client->settings->CompressionLevel = PACKET_COMPR_TYPE_RDP8;
+	return TRUE;
+}
+
+/**
+ * Handles an incoming client connection, to be run in it's own thread.
+ *
+ * arg is a pointer to a freerdp_peer representing the client.
+ */
+static DWORD WINAPI pf_server_handle_client(LPVOID arg)
+{
+	HANDLE eventHandles[32];
+	HANDLE ChannelEvent;
+	DWORD eventCount;
+	DWORD tmp;
+	DWORD status;
+	pServerContext* ps;
+	proxyData* pdata;
+	freerdp_peer* client = (freerdp_peer*) arg;
+
+	if (!init_p_server_context(client))
+	{
+		freerdp_peer_free(client);
+		return 0;
+	}
+
+	ps = (pServerContext*) client->context;
+	ps->dynvcReady = CreateEvent(NULL, TRUE, FALSE, NULL);
+	pdata = calloc(1, sizeof(proxyData));
+	ps->pdata = pdata;
+	/* inject configuration to proxyData */
+	pdata->config = client->ContextExtra;
+	client->settings->SupportGraphicsPipeline = TRUE;
+	client->settings->SupportDynamicChannels = TRUE;
+	/* TODO: Read path from config and default to /etc */
+	client->settings->CertificateFile = _strdup("server.crt");
+	client->settings->PrivateKeyFile = _strdup("server.key");
+	client->settings->RdpKeyFile = _strdup("server.key");
+
+	if (!client->settings->CertificateFile || !client->settings->PrivateKeyFile
+	    || !client->settings->RdpKeyFile)
+	{
+		WLog_ERR(TAG, "Memory allocation failed (strdup)");
```

Comment:
change the log so it mentions `CertificateFile`  `PrivateKeyFile` etc..

Reply:
go for it